contracts/Delegator.sol

```diff
@@ -27,10 +45,35 @@ contract Delegator is ACL {
             string memory url
         )
     {
-        return assetManager.getJob(id);
+        return assetManager.getJob(ids[_name]);
     }
 
-    // function getResult(uint8 id) public view returns (uint256) {
-    //     return assetManager.getResult(id);
-    // }
+    function getCollection(bytes32 _name)
+        external
+        view
+        returns (
+            bool active,
+            int8 power,
+            uint8[] memory jobIDs,
+            uint32 aggregationMethod,
+            string memory name
+        )
+    {
+        return assetManager.getCollection(ids[_name]);
+    }
+
+    function getNumActiveAssets() external view returns (uint256) {
+        return (assetManager.getActiveAssets()).length;
+    }
+
+    function getActiveAssets() external view returns (uint8[] memory) {
+        return assetManager.getActiveAssets();
+    }
+
+    function getResult(bytes32 _name) public view returns (uint32) {
```

Comment:
Hey is this possible scenario ?

Epoch x
ActiveAssets [1,2,3]
BlockMedian [10,20,30]

Epoch X + 1
State Confirm : Deactivate 2, ActiveAssets[1,3]
Client calls get result for asset id 3, index we get is 1
when we fetch median from lastBlock we fetch it for 1, where as in last block it was at 2
giving incorrect value

Reply:
Now thinking about this, the following scenario can be possible